internal/models/pubkey_resource.go

```diff
@@ -9,23 +9,27 @@ type PubkeyResource struct {
 	// Required auto-generated PK.
 	ID uint64 `db:"id" json:"id"`
 
-	// Auto-generated random string, not unique, not returned through API, read-only.
+	// Auto-generated random string, not DB-unique but pseudo-unique (128 bits),
+	// not returned through API, read-only. Used to "tag" resources in clouds for
+	// later use (e.g. search a resource by a tag). Use FormattedTag function to
+	// format it with a proper prefix.
 	Tag string `db:"tag" json:"-"`
 
 	// Associated Account model. Required.
 	PubkeyID uint64 `db:"pubkey_id" json:"pubkey_id"`
 
 	// Provider constant (for example ProviderTypeAWS). Required.
-	Provider int `db:"provider" json:"provider"`
+	Provider ProviderType `db:"provider" json:"provider"`
 
 	// Resource handle (id). Format is provider-dependant. Required.
 	Handle string `db:"handle" json:"handle"`
 }
 
-// FormattedTag returns ID and Tag concatenated in a safe way for clouds. That means
+// FormattedTag returns Tag concatenated in a safe way for clouds. That means
 // it does not start with a number, only includes alpha-num characters and dash.
+// Tag is
 func (p *PubkeyResource) FormattedTag() string {
-	return fmt.Sprintf("pk-%d-%s", p.ID, p.Tag)
+	return fmt.Sprintf("pk-%s", p.Tag)
 }
```

Comment:
I am actually changing the tag format from pk-ID-TAG to just pk-TAG. I think the probability of getting same TAG twice is low. I am increasing length to 22 characters which is in base62 sligtly bigger entropy than UUID. Chances are low.

Because I have a chicken-and-egg problem - do I create resource in the DB first (thus getting ID and Tag), upload and then update handle or the other way? Its complicated, I think its much easier to simply generate good enough TAG and live with it.

Reply:
makes sense, lets hope we won't be unlucky ^_^ it also avoid exposing internal ids, which might have caused some traction with security team :)

Reply:
You've not changed the test for this method tho ;)